packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -5,21 +5,43 @@
 
 import React from "react";
 
-import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
 import {
+	handleIncomingMessage,
+	ISourcedDebuggerMessage,
 	InboundHandlers,
 	TelemetryHistoryMessage,
-	GetTelemetryHistoryMessage,
 	TelemetryEventMessage,
-	ISourcedDebuggerMessage,
-	handleIncomingMessage,
-	postMessagesToWindow,
 } from "@fluid-tools/client-debugger";
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { useMessageRelay } from "../MessageRelayContext";
+
+function mapEventCategoryToBackgroundColor(eventCategory: string): string {
+	switch (eventCategory) {
+		case "generic":
+			return "#b8ebf2";
+		case "performance":
+			return "#4cf5a3";
+		case "error":
+			return "#f54c4f";
+		default:
+			return "#d2d3d4";
+	}
+}
+
+function replacer(key, value): unknown {
+	// Filtering out properties
+	if (key === "eventName" || key === "category") {
+		return undefined;
+	}
+	return value;
+}
 
 /**
  * Displays telemetry events generated by FluidFramework in the application.
  */
 export function TelemetryView(): React.ReactElement {
+	const messageRelay = useMessageRelay();
+
 	const [telemetryEvents, setTelemetryEvents] = React.useState<ITelemetryBaseEvent[]>([]);
```

Comment:
It might be worth initializing this to `undefined`, and displaying a spinner until we get a response from the debugger. That way we can differentiate the "I haven't received my data yet" from "There have been no telemetry logs yet" cases.